ego4d/internal/university_files.py

```
@@ -0,0 +1,721 @@
+                    f"Video: {video_id} points to participant file at"
+                    " non-existent participant_id: "
+                    f"'{video_metadata.recording_participant_id}'\n"
+                )
+            if video_metadata.recording_participant_id is None:
+                error_message += (
+                    f"Video: {video_id} does not have corresponding"
+                    " participant_id \n"
+                )
+
+        #   2. scenario_ids are in scenarios
+        for scenario_id in video_metadata.video_scenario_ids:
+            if scenario_id not in scenarios:
+                error_message += (
+                    f"Video: {video_id} points to scenarios at"
+                    f" non-existent scenario_id: '{scenario_id}'\n"
+                )
+
+        #   3. device_ids are in devices
+        if video_metadata.device_id and video_metadata.device_id not in devices:
+            error_message += (
+                f"Video: {video_id} points to devices at"
+                f" non-existent device_id: '{video_metadata.device_id}'\n"
+            )
+
+        #   4. physical_settings are in physical_setting
+        if physical_setting_dict:
+            if (
+                video_metadata.physical_setting_id
+                and video_metadata.physical_setting_id not in physical_setting_dict
+            ):
+                error_message += (
+                    f"Video: {video_id} points to physical setting at"
+                    " non-existent physical_setting_id: "
+                    f"'{video_metadata.physical_setting_id}'\n"
+                )
+
+        #   5. university_video_ids are in components
+        if video_id not in video_components_dict:
+            error_message += (
+                f"Video: {video_id} points to component at"
+                " non-existent university_video_id: "
+                f"'{video_id}'\n"
+            )
+
+    if error_message:
+        print(error_message)
```

Let's return the error messages and save them to disk as well as printing them to stdout